src/supervise-daemon/supervise-daemon.c

```diff
@@ -924,9 +972,9 @@ int main(int argc, char **argv)
 			break;
 
 		case 'D':  /* --respawn-delay time */
-			n = sscanf(optarg, "%d", &respawn_delay);
-			if (n	!= 1 || respawn_delay < 1)
+			if (!parse_respawn(&respawn_delay, optarg))
 				eerrorx("Invalid respawn-delay value '%s'", optarg);
+			rc_service_value_set(svcname, "respawn_delay", optarg);
```

Comment:
This seems to be going in the opposite direction of https://github.com/OpenRC/openrc/issues/800